templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```
@@ -52,11 +52,18 @@ contents:
             "{{ onPremPlatformIngressIP . }}")
         DOMAIN="{{.DNS.Spec.BaseDomain}}"
         if [[ -n "$NAMESERVER_IP" ]]; then
-            source /etc/os-release
-            if [[ "$NAME" == "Fedora" ]]; then
-                echo "[Resolve]" > /etc/systemd/resolved.conf.d/60-kni.conf
-                echo "DNS=$NAMESERVER_IP" >> /etc/systemd/resolved.conf.d/60-kni.conf
-                echo "Domains=$DOMAIN" >> /etc/systemd/resolved.conf.d/60-kni.conf
+            if [[ "$(systemctl is-enabled systemd-resolved)" == "enabled" ]]; then
```

```suggestion
            if systemctl -q is-enabled systemd-resolved; then
```

It's safer to look at the exit code of `systemctl is-enabled systemd-resolved` rather than matching on `enabled` since other statuses could also mean the service is enabled. You can then use `-q` to suppress the output.

https://www.freedesktop.org/software/systemd/man/systemctl.html#is-enabled%20UNIT%E2%80%A6

---

Makes sense...  Let me give it a quick test on my build cluster